app/javascript/components/lists/ListGroup.js

```diff
@@ -0,0 +1,78 @@
+import React, { useState, useEffect } from "react"
+import ListEntry from "./ListEntry"
+
+const ListGroup = (props) => {
+  const [lists, setLists] = useState(props.lists);
+
+  const csrf = props.csrf;
+  const inputRef = React.useRef(null);
+
+  const handleStateChange = ( listId ) => {
+
+    var newLists = [...lists];
+
+    newLists.map((list, index) =>{
+      let id = list.id;
+      if( id === listId ){
+        newLists.splice(index, 1);
+        setLists(newLists);
+      }
+
+      if( newLists.length > 0 ){
+        let id = ("list-" + newLists[0].id + "-list");
+        document.getElementById(id).click();
+      } else {
+        window.location.reload();
+      }
+    })
+  }
+
+  const displayList = (list, index) => {
+    let activeClass = "";
+    if(index == 0){
+      activeClass = "active";
```

Comment:
This is presentation logic that should belong to the ListEntry. Instead of using the class string, what do you think about using a Boolean here and then the ListEntry can add the class if active is true?

Reply:
Sounds good! Will work on this.